src/components/basic_table/basic_table.tsx

```diff
@@ -884,7 +884,11 @@ export class EuiBasicTable<T = any> extends Component<
     const cells = [];
 
     const { itemId: itemIdCallback } = this.props;
-    const itemId: ItemIdResolved = getItemId(item, itemIdCallback) || rowIndex;
+    const itemId: ItemIdResolved =
+      getItemId(item, itemIdCallback) !== null &&
+      typeof getItemId(item, itemIdCallback) !== 'undefined'
```

Comment:
```suggestion
      getItemId(item, itemIdCallback) != null
```

Will catch `null` and `undefined` but not `0`